File: Course Materials/Examples and Templates/OpenMP_cours/cours_ex5.c
#include <stdio.h>
#include <omp.h>

/* Dans un sous-programme appelé dans une région parallèle, 
les variables locales et automatiques sont implicitement privées
 à chacune des tâches (elles sont définies dans la pile de chaque tâche).
*/

int main()
{
  void sub(void);

#pragma omp parallel default(shared)
  {
    sub();
  }
  return 0;
}



void sub(void)
{
  int a;

  a=92290;
  a = a + omp_get_thread_num();
  printf("a vaut : %d\n", a);
}

File: Course Materials/Examples and Templates/OpenMP_cours/cours_ex5.f90
program parallel

! Dans un sous-programme appelé dans une région parallèle, 
!les variables locales et automatiques sont implicitement privées
! à chacune des tâches (elles sont définies dans la pile de chaque tâche).


 implicit none

 !$OMP PARALLEL DEFAULT(SHARED)
    call sub()
 !$OMP END PARALLEL
end program parallel


subroutine sub()
 use OMP_LIB
 implicit none
 integer :: a

  a = 92290
  a = a + OMP_GET_THREAD_NUM()
  print *,"A vaut : ",a
end subroutine sub
